### PR TITLE
Show information about current revision if not latest

### LINKED ISSUE #1945: Lack of information in (and about) revision pages
In revision pages, some informations are missing or hard to find:
- timestamp
- contributor name
- revision content (ex. `Selected images (Changed: nutrition_fr) - new image nutrition_fr : 25.72`)
- revision source
- the link to the last revision of the product

To find these informations you have to go to "Changes history" at the bottom and find which entry match to current revision (with mouse over "view" link).

Also, for a beginner, it's difficult to understand that this is an old version of the product.

Related to this issue, each "Changes history" entry is poor:
`<li><time datetime="2018-06-10T15:06:20">June 10, 2018 at 3:06:20 PM CEST</time> - <a href="/contributor/kiliweb">kiliweb</a> Nutriments (Changed: energy, fat, saturated-fat, proteins)  - (app)User : UUpsZkc3WUIvLzlSdmZZQndSZkY2K0p6ekxhcUF6bUdjN0UrSVE9PQ - <a href="/product/3095756220014/fleuron-de-canard-fleury-michon?rev=40">view</a></li>`

I suggest either or both:
- show these informations at the beginning of the page, in a way that clearly shows that this is an old revision; this informations should be presented with semantic code (example below)
- modify "Changes history" entries to let designers and coders better use "Changes history" informations

**Example of code for warning revision block**
```
<div id="rev_summary">
<span class="rev_warning">This page is an old revision this product.</span> 
Rev. <span class="rev_nb">40</span> 
produced on <time datetime="2018-06-10T15:06:20">June 10, 2018 at 3:06:20 PM CEST</time> 
by <a href="/contributor/kiliweb" class="rev_contributor">kiliweb</a>, 
who changed <span class="rev_changes">Nutriments (Changed: energy, fat, saturated-fat, proteins)</span> 
from source <span class="rev_source">(app)User : UUpsZkc3WUIvLzlSdmZZQndSZkY2K0p6ekxhcUF6bUdjN0UrSVE9PQ</span>.<br/>
Link to this revision: <a href="/product/3095756220014/fleuron-de-canard-fleury-michon?rev=40">view</a><br/>
Link to last revision: <a href="/product/3095756220014/fleuron-de-canard-fleury-michon">view</a>
</div>
```

**Example of Changes history entry**
```
<li id="rev_40" class="rev_set"><!-- <= add class="current" if it matches current revision -->
Rev. <span class="rev_nb">40</span> - 
<time datetime="2018-06-10T15:06:20">June 10, 2018 at 3:06:20 PM CEST</time> - 
<a href="/contributor/kiliweb" class="rev_contributor">kiliweb</a> 
<span class="rev_changes">Nutriments (Changed: energy, fat, saturated-fat, proteins)</span> - 
<span class="rev_source">(app)User : UUpsZkc3WUIvLzlSdmZZQndSZkY2K0p6ekxhcUF6bUdjN0UrSVE9PQ</span> - 
<a class="rev_link">href="/product/3095756220014/fleuron-de-canard-fleury-michon?rev=40">view</a>
</li>`
```

Solving this issue will help solve #1408.

### PR DESCRIPTION
**Pull request by VaiTon**
Partially fixes #1945

### CODE REVIEW

**Comment by VaiTon** [comment]
image

**Comment by CharlesNepote** [comment]
Great! As in wikipedia, would you use a symbol to make a visual separation between previous version, latest version and next version? Wikipedia uses a pipe (|).

image

**Comment by VaiTon** [comment]
> 
> 
> Great! As in wikipedia, would you use a symbol to make a visual separation between previous version, latest version and next version? Wikipedia uses a pipe (|).
> 
> image

Is this really necessary? I really like its look without any symbol between links. XD

**Comment by VaiTon** [comment]
The last thing i would like to add is the comment and the changes.

File: lib/ProductOpener/Display.pm [hunk 1]
```diff
@@ -6797,8 +6796,10 @@ $Lang{warning_gs1_company_prefix}{$lc}
 HTML
 ;
 	}
-
-	if (not has_tag($product_ref, "states", "en:complete")) {
+	if (defined $rev) {
+		$html .= display_rev_info($code, $rev);
+	}
+	if (not has_tag($product_ref, "states", "en:complete") and not defined $rev) {
```
Comment (stephanegigandet) [inline lib/ProductOpener/Display.pm]:
```suggestion
	elsif (not has_tag($product_ref, "states", "en:complete")) {
```
I'm always in doubt about precedence when I see things "like and not defined" without parenthesis.

**Comment by VaiTon** [comment]
@stephanegigandet @hangy from TravisCI
```
Package graphviz is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
```

**Pull request merged**